### DISCUSSION TITLE
How to ComfyUI on AMD Radeon RX 5700 XT

### DISCUSSION THREAD

**Discussion by Robin-Aschan** [comfyanonymous/ComfyUI#1119] [Q&A] [answered]
Did anyone get ComfyUI to work on the AMD 5700 XT card on linux? When I follow the instructions I get errors. I know people have got it to work on AMD cards of other version, e.g. 6XXX models and above.

**Comment by jacobhawkins495** [comment, marked as answer]
Not sure what errors you are getting, but here are the steps I took to get ComfyUI to run on my Radeon RX 5700 XT under Arch Linux:

1. In a terminal, run `git clone https://github.com/comfyanonymous/ComfyUI.git`
2. In a terminal, run `cd ComfyUI`
3. Put your Stable Diffusion model files in `ComfyUI/models/checkpoints`
4. Put your VAE model files into `ComfyUI/models/vae` if you have any
5. In a terminal, run `virtualenv -p 3.10 sdxl` (This creates a Python virtual environment to prevent this from clogging up your main system Python since we need specific versions of packages to make ComfyUI work)
6. In a terminal, run `source sdxl/bin/activate` (This activates the Python virtual environment. Must be done every time you launch ComfyUI from a new terminal window)
7. In a terminal, run `pip install torch==1.13.1+rocm5.2 torchvision==0.14.1+rocm5.2 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/rocm5.2`. (Note that this is different from the official installation steps. ROCm 5.4.2/Torch 2.0+ doesn't run on the RX 5700 XT, we need this older version for our technically unsupported cards.)
8. In a terminal, do `pip install -r requirements.txt`
9. In a terminal, run `HSA_OVERRIDE_GFX_VERSION=10.3.0 python main.py --force-fp32`. ComfyUI should launch and you should be able to access the UI in your web browser at `http://127.0.0.1:8188` after it finishes loading. (look in the terminal for the message `To see the GUI, go to http://127.0.0.1:8188` to know when it has finished loading) 
10. Enter your prompt and negative prompt, set your image size and batch settings, and hit `Queue Prompt`. Your GPU fan should kick on after a minute or so and the UI should start indicating progress (Look for a green outline moving around the screen and after a minute or two a progress bar should show up in the terminal and in the box titled `KSampler`)

After all that, ComfyUI should output an image to the `ComfyUI/outputs` directory. To launch ComfyUI subsequent times, follow step 6, then step 9. My RX 5700 XT can output an image in 2-6 minutes using Stable Diffusion XL, so if your image is taking 10+ minutes to generate or the progress bar never shows up, something is not working correctly.

**Reply by TerminatorHR** [reply]
@jacobhawkins495 Thank you. I also needed to do 7.5 step "pip install -r requirements.txt". 

Is 5700 XT still limited to these older versions in step 7?

**Reply by jacobhawkins495** [reply]
Whoops, added the missing step, thanks!

Unfortunately there was a [regression in ROCm version 5.3](https://github.com/RadeonOpenCompute/ROCm/issues/2527) that broke support for the 5700XT and it has never been fixed, which means no version of ROCm newer than 5.2 supports the 5700XT. If this regression does ever get fixed, installing the older versions will no longer be needed.

**Reply by arunhk3** [reply]
Some people built Torch and Torchvision which works fine with RX 5700XT.

Below are the links if anyone wants to install and use:

**Torch:** https://repo.radeon.com/rocm/manylinux/.private-9bd8754a636553e1d1ce3e107bf18c00/20231018/torch-2.1.0%2Brocm5.2-cp310-cp310-linux_x86_64.whl

**Torchvision:** https://github.com/kmsedu/pytorch_rocm52_wheels/raw/main/torchvision-0.17.0a0+rocm5.2-cp310-cp310-linux_x86_64.whl
